xqueue_watcher/client.py

```diff
@@ -158,11 +158,19 @@ def process_one(self):
                 self.processing = True
                 success = self._handle_submission(content)
             return success
-        except requests.exceptions.Timeout:
-            return True
+        except requests.exceptions.Timeout as e:
+            request_body_summary = ''
+            if e.request.body:
+                # self._request uses Content-Type: 'application/x-www-form-urlencoded', which can be
+                # decoded using urlparse.parse_qs
+                request_body_summary = (
+                    "(xqueue_header: %s)" % urlparse.parse_qs(e.request.body).get('xqueue_header', '')
+                )
+            log.error("Timeout during request to '%s' %s", e.request.url, request_body_summary)
+            return False
         except Exception as e:
             log.exception(e.message)
-            return True
+            return False
```

Comment:
I'm assuming that the false response is what triggers the wait before processing the next submission?

Reply:
That's right: https://github.com/mitodl/xqueue-watcher/pull/2/files/4f2d8bdd1b0b011ef248ff7795f3f61f517a7686#diff-d1f565998a9574b4c6b22e7fdc0552cfR195